File: 프로그래머스/unrated/133502. 햄버거 만들기/햄버거 만들기.py
def solution(ingredient):
    q = []
    answer = 0
    for i in ingredient:
        q.append(i)
        
        if len(q) >= 4 and q[-4:] == [1, 2, 3, 1]:
            answer += 1
            q.pop()
            q.pop()
            q.pop()
            q.pop()
            
    return answer

File: 프로그래머스/unrated/133502. 햄버거 만들기/햄버거 만들기.cs
using System;
using System.Collections.Generic;

public class Solution {
    public int solution(int[] ingredient) {
        int answer = 0;
        List<int> temp = new List<int>();
        
        foreach (int v in ingredient)
        {
            temp.Add(v);
            
            if (temp.Count < 4)
                continue;
            
            if (temp[temp.Count - 4] == 1 && temp[temp.Count - 3] == 2 && temp[temp.Count - 2] == 3 && temp[temp.Count - 1] == 1)
            {
                temp.RemoveRange(temp.Count - 4, 4);
                answer += 1;
            }
        }
        return answer;
    }
}